docs: add user manual for end-user contribution workflow (#1528)
finos/git-proxy: website/docs/user-manual.mdx

- **Proxy authentication (for git operations):** You authenticate using a **Personal Access Token (PAT)** from the target SCM platform (e.g., GitHub PAT). GitProxy forwards these credentials when cloning repositories to calculate diffs and when pushing approved changes upstream. For GitHub you'll need a personal access token (classic) with `repo` permissions. See [GitHub's documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) for details.
- **UI authentication (for the web interface):** GitProxy supports multiple authentication methods for the web interface, including local accounts, Active Directory/LDAP, and OpenID Connect (OIDC). See the [Authentication configuration](/docs/configuration/authentication) page for details on available options.

:::info Your organisation's configuration
The specific authentication method used for the GitProxy web interface depends on how your organisation has deployed GitProxy. Contact your OSPO or Open Source leads for the correct login credentials and method.
:::

#### Checks applied to contributions

> GitProxy applies a series of automated checks (called processors) to every push. These checks run in sequence, and if any check fails, the push is blocked. The main checks include:

Reviewer: Added link to the `Processors.md` guide for more details:

```suggestion
GitProxy applies a series of automated checks (called [Processors](processors.md)) to every push. These checks run in sequence, and if any check fails, the push is blocked. The main checks include:
```

Author: If we merge the 'unifying' PR of yours first this can change to an internal link

> GitProxy applies a series of automated checks (called [Processors](processors.md)) to every push. These checks run in sequence, and if any check fails, the push is blocked. The main checks include:

Author: ```suggestion
GitProxy applies a series of automated checks (called [Processors](architecture/processors.md)) to every push. These checks run in sequence, and if any check fails, the push is blocked. The main checks include:
```

**Review process:**

1. **Log in to GitProxy:** Navigate to your GitProxy web interface and authenticate
2. **View pending contributions:** Go to the "Pushes" or "Dashboard" tab to see contributions awaiting review
3. **Examine the contribution:** Click on a pending push to view:
   - Full diff of all changes
   - Commit messages and author information
   - Results of automated security and quality checks
   - Repository and branch information
4. **Complete the attestation:** Before approving, you must confirm the contribution complies with policies and is appropriate for the target repository
5. **Make a decision:**
   - **Approve:** Click "Approve". The contributor can then push again to send changes upstream.
   - **Reject:** Click "Reject" and provide clear feedback explaining what needs to be fixed.

**Reviewer responsibilities:**

- Ensure contributions don't contain sensitive organisational information or intellectual property
- Verify changes align with the target project's contribution guidelines
- Check that code quality meets acceptable standards
- Confirm no credentials, secrets, or PII are being exposed
- Provide constructive feedback when rejecting contributions

> ### Troubleshooting

Reviewer: I'm wondering if we could specify the expected error *messages* here, similar to the [troubleshooting guide for the v2 upgrade](https://github.com/finos/git-proxy/blob/main/docs/Upgrading%20to%20v2.md).

It'll help users identify their problem faster, and also improve the likelihood of finding this doc page when they (or their AI agents) Google the error message.

Author: The credentials prompts are usuabllly UI prompts and can vary - VS code (and derivates) will catch the request and display their own UI, git bash will display Git Credentials Manager popups etc.. Hence, prob best to keep the firs issue as is

> - Provide constructive feedback when rejecting contributions
> 
> ### Troubleshooting
> 
> **Git keeps asking for my credentials**
> 
> If you're prompted for credentials on every git operation, the Git Credential Manager (GCM) may be disabled or misconfigured.
> 
> Verify if your GCM configuration has been altered:

Author: ```suggestion
**Git keeps asking for my credentials**

If you're prompted for credentials on every git operation, the Git Credential Manager (GCM) may be disabled or misconfigured.

:::info
Credentials request popups will vary based on your git client. VS code (and derivates) may catch the credentials request and display their own UI, while Git Bash will display Git Credentials Manager popups.
:::

Verify if your GCM configuration has been altered:
```
